File: trunk/lucette2/src/main/java/battle/csn/lucette/lan/ConnectionFactory.java
package battle.csn.lucette.lan;

import java.io.IOException;
import java.net.HttpURLConnection;
import java.net.InetSocketAddress;
import java.net.Proxy;
import java.net.URL;

import com.sun.jersey.client.urlconnection.HttpURLConnectionFactory;


/**
 * Connexion derrière un proxy.
 * @author jgoulley
 * 
 */
public class ConnectionFactory implements HttpURLConnectionFactory
{
    Proxy proxy;

    private void initializeProxy( )
    {
        proxy = new Proxy( Proxy.Type.HTTP, new InetSocketAddress( "ptx.proxy.corp.sopra", 8080 ) );
    }

    public HttpURLConnection getHttpURLConnection( URL url ) throws IOException
    {
        initializeProxy( );
        return (HttpURLConnection) url.openConnection( proxy );
    }
}

File: trunk/lucette2/src/main/scala/battle/csn/lucette2/network/ConnectionFactory.scala
package battle.csn.lucette2.network

import java.net.URL
import java.net.Proxy
import java.net.InetSocketAddress

class ConnectionFactory {
    var proxy:Proxy =_

    def initializeProxy( )
    {
        proxy = new Proxy( Proxy.Type.HTTP, new InetSocketAddress( "ptx.proxy.corp.sopra", 8080 ) )
    }

    def getHttpURLConnection(  url :URL)=
    {
        initializeProxy( )
        url.openConnection( proxy )
    }
}